tests/test_extract_contained_procedures.py

```diff
@@ -0,0 +1,502 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import pytest
+from loki.sourcefile import Sourcefile
+from loki.expression import FindVariables, FindInlineCalls
+from loki.ir import (
+    CallStatement, Import,
+)
+from loki.visitors import FindNodes
+from loki.transform import (
+    extract_contained_procedures
+)
+from loki.subroutine import Subroutine
+
+def test_extract_contained_procedures_basic_scalar():
```

Comment:
To ensure compatibility across frontends, I suggest to parametrize the tests in the usual way, see for example: 
https://github.com/ecmwf-ifs/loki/blob/955a9da0bed33ec04e9578650164842543652cf6/tests/test_transform_inline.py#L34-L35

And then specify the frontend in the Sourcefile constructor, as is done for the module constructor here:
https://github.com/ecmwf-ifs/loki/blob/955a9da0bed33ec04e9578650164842543652cf6/tests/test_transform_inline.py#L66

Reply:
Thanks. Tests now run for all frontends.